tests/subsys/retention/boards/nucleo_f746zg.overlay

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ &i2c1 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+
+	pcal6416a: pcal6416a@20 {
+		compatible = "nxp,pcal6416a";
+		status = "okay";
+		reg = <0x20>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <16>;
+		int-gpios = <&gpioc 6 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		reset-gpios = <&gpiob 15 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		retention-driver-state = <&pcal6416a_driver_state>;
+	};
+};
+
+&dtcm {
```

Comment:
Not sure where dtcm is defined but generally you allocate a small portion for retained memory usage e.g.:
```
        sram@2003FC00 {
                compatible = "zephyr,memory-region", "mmio-sram";
                reg = <0x2003FC00 DT_SIZE_K(1)>;
                zephyr,memory-region = "RetainedMem";
                status = "okay";

                retainedmem {
                        compatible = "zephyr,retained-ram";
                        status = "okay";
                        #address-cells = <1>;
                        #size-cells = <1>;
```

Reply:
I wanted to keep the test as simple as possible, therefore I've used the whole DTCM for the retention. But I also do not mind cutting some space of at the end for this purpose.